File: log-analysis-frontend/src/api/datasource.ts
import request from '@/utils/request'

/**
 * 数据源类型
 */
export interface Datasource {
  id: string
  name: string
  type: string
  host: string
  port: number
  databaseName?: string
  username?: string
  password?: string
  sslEnabled?: boolean
  connectionParams?: string
  description?: string
  status: string
  lastCheckTime?: string
  lastCheckStatus?: string
  lastCheckMessage?: string
  createdAt: string
  updatedAt: string
}

/**
 * 创建数据源请求
 */
export interface CreateDatasourceRequest {
  name: string
  type: string
  host: string
  port: number
  databaseName?: string
  username?: string
  password?: string
  sslEnabled?: boolean
  connectionParams?: string
  description?: string
}

/**
 * 更新数据源请求
 */
export interface UpdateDatasourceRequest {
  name?: string
  host?: string
  port?: number
  databaseName?: string
  username?: string
  password?: string
  sslEnabled?: boolean
  connectionParams?: string
  description?: string
  status?: string
}

/**
 * 数据源测试结果
 */
export interface DatasourceTestResult {
  success: boolean
  message: string
  responseTime?: number
  version?: string
  details?: string
}

/**
 * 分页查询数据源列表
 */
export function listDatasources(params: {
  pageNum?: number
  pageSize?: number
  keyword?: string
  type?: string
  status?: string
}) {
  return request({
    url: '/api/datasources',
    method: 'get',
    params
  })
}

/**
 * 获取所有活跃的数据源
 */
export function listActiveDatasources() {
  return request({
    url: '/api/datasources/active',
    method: 'get'
  })
}

/**
 * 根据类型查询数据源
 */
export function listDatasourcesByType(type: string) {
  return request({
    url: `/api/datasources/by-type/${type}`,
    method: 'get'
  })
}

/**
 * 获取数据源详情
 */
export function getDatasource(id: string) {
  return request({
    url: `/api/datasources/${id}`,
    method: 'get'
  })
}

/**
 * 创建数据源
 */
export function createDatasource(data: CreateDatasourceRequest) {
  return request({
    url: '/api/datasources',
    method: 'post',
    data
  })
}

/**
 * 更新数据源
 */
export function updateDatasource(id: string, data: UpdateDatasourceRequest) {
  return request({
    url: `/api/datasources/${id}`,
    method: 'put',
    data
  })
}

/**
 * 删除数据源
 */
export function deleteDatasource(id: string) {
  return request({
    url: `/api/datasources/${id}`,
    method: 'delete'
  })
}

/**
 * 测试数据源连接
 */
export function testDatasourceConnection(id: string) {
  return request({
    url: `/api/datasources/${id}/test`,
    method: 'post'
  })
}

/**
 * 测试新数据源连接（创建前测试）
 */
export function testNewDatasourceConnection(data: CreateDatasourceRequest) {
  return request({
    url: '/api/datasources/test',
    method: 'post',
    data
  })
}

File: log-analysis-backend/log-analysis-app/src/main/java/cn/mw/loganalysis/datasource/entity/Datasource.java
package cn.mw.loganalysis.datasource.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 数据源实体
 */
@Data
@TableName("datasources")
public class Datasource {

    /**
     * 数据源ID（UUID）
     */
    @TableId(type = IdType.ASSIGN_UUID)
    private String id;

    /**
     * 数据源名称（唯一）
     */
    @TableField("name")
    private String name;

    /**
     * 数据源类型: clickhouse, elasticsearch, postgresql, mysql, loki
     */
    @TableField("type")
    private String type;

    /**
     * 主机地址
     */
    @TableField("host")
    private String host;

    /**
     * 端口号
     */
    @TableField("port")
    private Integer port;

    /**
     * 数据库名称
     */
    @TableField("database_name")
    private String databaseName;

    /**
     * 用户名
     */
    @TableField("username")
    private String username;

    /**
     * 密码（加密存储）
     */
    @TableField("password")
    private String password;

    /**
     * 是否启用 SSL
     */
    @TableField("ssl_enabled")
    private Boolean sslEnabled;

    /**
     * 额外连接参数（JSON 格式）
     */
    @TableField("connection_params")
    private String connectionParams;

    /**
     * 数据源描述
     */
    @TableField("description")
    private String description;

    /**
     * 状态: active, inactive, error
     */
    @TableField("status")
    private String status;

    /**
     * 最后检查时间
     */
    @TableField("last_check_time")
    private LocalDateTime lastCheckTime;

    /**
     * 最后检查状态: success, failed
     */
    @TableField("last_check_status")
    private String lastCheckStatus;

    /**
     * 最后检查消息
     */
    @TableField("last_check_message")
    private String lastCheckMessage;

    /**
     * 创建时间
     */
    @TableField("created_at")
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField("updated_at")
    private LocalDateTime updatedAt;

    /**
     * 创建人
     */
    @TableField("created_by")
    private Long createdBy;

    /**
     * 更新人
     */
    @TableField("updated_by")
    private Long updatedBy;
}
